### PR TITLE
Add optional module telemetry

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## Overview/Summary

PR to add optional module telemetry using the ARM template method.

## Testing Evidence

image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located in the [Enterprise-Scale repo](https://github.com/Azure/Enterprise-Scale) in the directory: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: locals.telemetry.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"
+  "contentVersion": "1.0.0.0",
+  "parameters": {},
+  "variables": {},
+  "resources": [],
+  "outputs": {
+    "telemetry": {
+      "type": "string",
+      "value": "For more information, see https://aka.ms/alz-terraform-module-telemetry"
```
Comment (krowlandson) [inline locals.telemetry.tf:78]: For consistency, do we want to shorten the quick link to:
https://aka.ms/alz/tf/telemetry

I've set this up ready.

File: docs/wiki/[Variables]-disable_telemetry.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+## Overview
```
Comment (krowlandson) [inline docs/wiki/[Variables]-disable_telemetry.md:1]: Do we want to update this with more context similar to the following page from our ALZ-Bicep Wiki?

https://github.com/Azure/ALZ-Bicep/wiki/CustomerUsage

Or duplicate this page for our Wiki with the necessary updates?

**Review by krowlandson — CHANGES_REQUESTED** [review]
Looking good... a couple of feedback items added inline.

Also, do we still want to go with the bitwise in the deployment ID, or have a couple of Subscription scoped deployments for `connectivity` and `management`? This might give us greater flexibility in the future, and make it easier to identify things like hub and spoke vs. vwan deployments, etc.

File: docs/wiki/[Variables]-disable_telemetry.md (hunk 2 above)
Comment (matt-FFFFFF) [reply docs/wiki/[Variables]-disable_telemetry.md:1]: Updated in latest commit - see what you think

**Comment by matt-FFFFFF** [comment]
> Looking good... a couple of feedback items added inline.
> 
> Also, do we still want to go with the bitwise in the deployment ID, or have a couple of Subscription scoped deployments for `connectivity` and `management`? This might give us greater flexibility in the future, and make it easier to identify things like hub and spoke vs. vwan deployments, etc.

Changed now, we have four deployments but identity is disabled for now until we add the provider declaration.

**Review by jtracey93 — APPROVED** [review]
LGTM 🚢

**Pull request merged**